osquery/main/windows/main.cpp

```diff
@@ -251,8 +251,9 @@ Status installService(const std::string& binPath) {
   HANDLE flagsFilePtr = nullptr;
   auto binPathWithFlagFile = binPath + " --flagfile=";
   auto flagsFile = FLAGS_flagfile.empty() ? kDefaultFlagsFile : FLAGS_flagfile;
-  binPathWithFlagFile += flagsFile;
-  flagsFilePtr = CreateFile(flagsFile.c_str(),
+  // "Wrap" the flag file in the event there are spaces in the path
+  binPathWithFlagFile += "\"" + flagsFile + "\"";
```

Comment:
What if `FLAGS_flagfile` already contains wrapped `"`?

Can you explore of `boost::io::quoted` solves the issue?

Reply:
Hrm.... I might try something more hacky. Using `quoted` is escaping our backslashes, which breaks the derivation of the flag file :(

```
PS C:\Users\Nicholas\work\repos\osquery\build\windows10> sc.exe qc osqueryd                                                       [SC] QueryServiceConfig SUCCESS

SERVICE_NAME: osqueryd
        TYPE               : 10  WIN32_OWN_PROCESS
        START_TYPE         : 2   AUTO_START
        ERROR_CONTROL      : 1   NORMAL
        BINARY_PATH_NAME   : "C:\Program Files\osquery\osqueryd\osqueryd.exe" --flagfile="\\Program Files\\osquery\\osquery.flags"
        LOAD_ORDER_GROUP   :
        TAG                : 0
        DISPLAY_NAME       : osquery daemon service
        DEPENDENCIES       :
        SERVICE_START_NAME : LocalSystem
```